.circleci/config.yml

```diff
@@ -68,85 +62,151 @@ jobs:
-          root: ./dist/
-          paths: .
+    steps: *build-steps
 
   build-sdist:
     docker:
-      - image: circleci/python:3.9-buster
-
-    working_directory: ~/build
+      - image: circleci/python:3.9
 
     steps:
       - checkout
-
       - run:
           name: build sdist
           command: |
-            python -m venv build_env
-            . build_env/bin/activate
+            python -m venv env
+            . env/bin/activate
             pip install -r requirements.txt
             python setup.py sdist -d ./dist
+      - store_artifacts: *store-artifacts
+      - persist_to_workspace: *persist-to-workspace
+
+  build-windows:
+    parameters:
+      python-version:
+        type: string
+
+    executor:
+      name: win/default
+
+    environment:
+      <<: *global-environment
+      CIBW_PROJECT_REQUIRES_PYTHON: ~=<< parameters.python-version>>
+      CIBW_ARCHS_WINDOWS: AMD64
+
+    steps:
+      - checkout
+      - run:
+          name: build wheels
+          command: |
+            python -m pip install cibuildwheel==2.2.2
+            python -m cibuildwheel --output-dir dist
+      - store_artifacts: *store-artifacts
+      - persist_to_workspace: *persist-to-workspace
+
+  deploy-all:
+    docker:
+      - image: circleci/python:3.9
+
+    steps:
+      - attach_workspace:
+          at: dist
 
       - store_artifacts:
           path: ./dist
 
-      - persist_to_workspace:
-          root: ~/build/dist/
-          paths: .
+      - run:
+          name: deploy
+          command: |
+            python -m venv env
+            . env/bin/activate
+            python -m pip install twine
+            twine upload -u $PYPI_USERNAME -p $PYPI_PASSWORD --skip-existing ./dist/*
 
-  test-sdist:
+  test-airspeed-velocity:
     docker:
       - image: circleci/python:3.9
 
     steps:
       - checkout
+      - run:
+          name: install dependencies
+          command: |
+            python -m venv env
+            . env/bin/activate
+            pip install -r benchmarks/requirements.txt
+      - run:
+          name: check that our airspeed velocity tests run without failure
+          command: |
+            . env/bin/activate
+            asv machine --yes
+            asv run --quick --strict
 
-      - attach_workspace:
-          at: dist
+  test-doctest:
+    docker:
+      - image: circleci/python:3.7 # as of march 2019 RTD uses 3.7
 
+    steps:
+      - checkout
       - run:
-          name: install from sdist
+          name: install doxygen
+          command: sudo apt-get install doxygen
+      - run:
+          name: install dependencies
           command: |
             python -m venv env
             . env/bin/activate
-            pip install dist/dimod*.tar.gz
+            pip install -r requirements.txt
+            pip install -r docs/requirements.txt
+      - run:
+          name: build dimod
+          command: |
+            . env/bin/activate
+            python setup.py build_ext --inplace
```

Comment:
Perhaps install from `dist`? To simplify and speed-up?

Reply:
I think I am going to leave this alone for now. If we did that, it would depend on the build job and I like that the doctest often finishes first since it's one of the ones that fails most often.